libs/base/Data/List.idr

```diff
@@ -556,6 +595,33 @@ public export
 catMaybes : List (Maybe a) -> List a
 catMaybes = mapMaybe id
 
+||| Map with index parameter
+|||
+||| ```idris example
+||| mapI (*) [4,1,2,1]
+||| ```
+export
+mapI : (Nat -> a -> b) -> List a -> List b
+mapI f list = mapI' Z list where
+  mapI' : Nat -> List a -> List b
+  mapI' _ [] = []
+  mapI' i (x::xs) = f i x :: mapI' (S i) xs
+
```

Comment:
See `contrib`'s [`Data.List.Extra](https://github.com/idris-lang/Idris2/blob/a9ccf4db4f69677c5287a1278ab167247dac161d/libs/contrib/Data/List/Extra.idr#L11).

Reply:
oh didnt see that. should I add all the functions in there and in Data.Vect.Extra and do a new request?